File: array/1539_kth_missing_positive_number/1539_kth_missing_positive_number.py
from typing import List

class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        n = 0
        for i in range(1, arr[-1] + k + 1):
            if i not in arr:
                n += 1
            if n == k:
                return i


arr = [1,2,3,4]
res = Solution().findKthPositive(arr, 2)
print(res)

File: array/1539_kth_missing_positive_number/1539_kth_missing_positive_number.go
package main

import "fmt"

func findKthPositive(arr []int, k int) int {
	prev, curr, cnt, inc := 0, 0, 0, 0
	for _, curr = range arr {
		inc = curr - prev - 1
		if cnt+inc >= k {
			break
		}
		prev = curr
		cnt += inc
	}
	inc = k - cnt
	return prev + inc
}

func main() {
	arr := []int{1, 2, 3, 4}
	res := findKthPositive(arr, 2)
	fmt.Println(res)
}
